jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -355,7 +355,7 @@ public void configure(WebAppContext context) throws Exception
     public void deconfigure(WebAppContext context) throws Exception
     {
         //if we're not persisting the temp dir contents delete it
-        if (!context.isPersistTempDirectory())
+        if (!context.isPersistTempDirectory() && context.getTempDirectory() != null && context.getTempDirectory().exists())
```

Comment:
assign `getTempDirectory()` to a variable

Reply:
Done.